orchestrator/careplanservice/service.go

```diff
@@ -351,16 +351,25 @@ func (s *Service) handleBundle(httpRequest *http.Request, httpResponse http.Resp
 	var resultHandlers []FHIRHandlerResult
 	for entryIdx, entry := range bundle.Entry {
 		// Bundle.entry.request.url must be a relative URL with at most one slash (so Task or Task/1, but not http://example.com/Task or Task/foo/bar)
-		resourcePath := entry.Request.Url
+		requestUrl, err := url.Parse(entry.Request.Url)
+		if err != nil {
+			coolfhir.WriteOperationOutcomeFromError(err, op, httpResponse)
+			return
+		}
+		if requestUrl.IsAbs() {
+			coolfhir.WriteOperationOutcomeFromError(fmt.Errorf("bundle.entry[%d].request.url (entry #) must be a relative URL", entryIdx), op, httpResponse)
+			return
+		}
+		resourcePath := requestUrl.Path
 		resourcePathParts := strings.Split(resourcePath, "/")
 		if entry.Request == nil || len(resourcePathParts) > 2 {
-			coolfhir.WriteOperationOutcomeFromError(fmt.Errorf("bundle.entry[%d].request.url (entry #) must be a relative URL", entryIdx), op, httpResponse)
+			coolfhir.WriteOperationOutcomeFromError(fmt.Errorf("bundle.entry[%d].request.url (entry #) has too many paths", entryIdx), op, httpResponse)
```

Comment:
The SAZ bundle (with url Patient?identifier=http......//...//../|111222333) will still crash now, or? (i.e. this does solve the problem....?)

Reply:
Yes, because all the stuff behind the ? is not part of the path:
```
requestUrl, err := url.Parse(entry.Request.Url)
```
Will parse the url `Patient?identifier=http......//...//../|111222333`
```
resourcePath := requestUrl.Path
```
Will result in `Patient`
```
resourcePathParts := strings.Split(resourcePath, "/")
```
Wil get resourcePathParts[0] only